container_service_extension/configure_cse.py

```diff
@@ -95,19 +97,41 @@ def check_cse_installation(config, msg_update_callback=utils.NullPrinter()):
                                             config['vcd']['password'])
         client.set_credentials(credentials)
 
-        # check that AMQP exchange exists
-        amqp = config['amqp']
-        credentials = pika.PlainCredentials(amqp['username'], amqp['password'])
-        parameters = pika.ConnectionParameters(amqp['host'], amqp['port'],
-                                               amqp['vhost'], credentials,
-                                               ssl=amqp['ssl'],
-                                               connection_attempts=3,
-                                               retry_delay=2, socket_timeout=5)
-        connection = None
-        try:
-            connection = pika.BlockingConnection(parameters)
-            channel = connection.channel()
+        if utils.use_mqtt_protocol(config):
+            # Check that MQTT extension exists
+            mqtt_ext_manager = MQTTExtensionManager(client)
```

Comment:
Can break down this code into functions for AMQP/MQTT specific operations?
i.e.

`
if utils.use_mqtt_protocol(config):
    do_things_specific_to_mqtt(ARGS)
`
`
else:
    do_things_specific_to_amqp(ARGS)
`
This way it will be more readable?

Right now the methods like check_cse_installation and install_cse are getting way to big.
Its hard to follow all the operations in long functions.

Reply:
Thank you, I have changed this.